cpp/CMakeLists.txt

```diff
@@ -246,46 +250,64 @@ file(INSTALL ${LIBCXX_INCLUDE_DIR}     DESTINATION ${CMAKE_BINARY_DIR}/include/l
 ###################################################################################################
 # - find zlib -------------------------------------------------------------------------------------
 
-find_package(ZLIB REQUIRED)
+CPMFindPackage(
+  NAME ZLIB
+  GITHUB_REPOSITORY madler/zlib
+  VERSION 1.2.11
+)
 
-message(STATUS "ZLIB: ZLIB_LIBRARIES set to ${ZLIB_LIBRARIES}")
-message(STATUS "ZLIB: ZLIB_INCLUDE_DIRS set to ${ZLIB_INCLUDE_DIRS}")
+message(STATUS "ZLIB: ZLIB_LIBRARY set to ${ZLIB_LIBRARY}")
+message(STATUS "ZLIB: ZLIB_INCLUDE_DIR set to ${ZLIB_INCLUDE_DIR}")
 
-if(ZLIB_FOUND)
-    message(STATUS "ZLib found in ${ZLIB_INCLUDE_DIRS}")
+if(ZLIB_INCLUDE_DIR)
+    message(STATUS "ZLib found in ${ZLIB_INCLUDE_DIR}")
 else()
     message(FATAL_ERROR "ZLib not found, please check your settings.")
-endif(ZLIB_FOUND)
+endif(ZLIB_INCLUDE_DIR)
+
 
 ###################################################################################################
 # - find boost ------------------------------------------------------------------------------------
 
 # Don't look for a CMake configuration file
 set(Boost_NO_BOOST_CMAKE ON)
 
-find_package(
-    Boost REQUIRED MODULE
-    COMPONENTS filesystem
+CPMFindPackage(
+  NAME Boost
+  GITHUB_REPOSITORY Orphis/boost-cmake
+  VERSION 1.67.0
+  FIND_PACKAGE_ARGUMENTS "COMPONENTS filesystem"
 )
 
-message(STATUS "BOOST: Boost_LIBRARIES set to ${Boost_LIBRARIES}")
-message(STATUS "BOOST: Boost_INCLUDE_DIRS set to ${Boost_INCLUDE_DIRS}")
+message(STATUS "BOOST: Boost_LIBRARY set to ${Boost_LIBRARY}")
+message(STATUS "BOOST: Boost_INCLUDE_DIR set to ${Boost_INCLUDE_DIR}")
 
-if(Boost_FOUND)
-    message(STATUS "Boost found in ${Boost_INCLUDE_DIRS}")
+if(Boost_INCLUDE_DIR)
+    message(STATUS "Boost found in ${Boost_INCLUDE_DIR}")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DBOOST_NO_CXX14_CONSTEXPR")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBOOST_NO_CXX14_CONSTEXPR")
     set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -DBOOST_NO_CXX14_CONSTEXPR")
 else()
     message(FATAL_ERROR "Boost not found, please check your settings.")
-endif(Boost_FOUND)
+endif(Boost_INCLUDE_DIR)
 
 ###################################################################################################
 # - RMM -------------------------------------------------------------------------------------------
 
 find_path(RMM_INCLUDE "rmm"
           HINTS "$ENV{RMM_ROOT}/include")
 
+# RMM fails to provide the RMM_SOURCE_DIR variable when installed with CPM. This does not work.
+if(NOT RMM_INCLUDE)
+  CPMFindPackage(
+    NAME RMM
+    GITHUB_REPOSITORY rapidsai/rmm
+    VERSION 0.16.0a
+  )
+
+  set(RMM_INCLUDE "${RMM_SOURCE_DIR}")
+endif(NOT RMM_INCLUDE)
```

Comment:
Should be able to just use `CPMFindPackage` and users can override via `RMM_ROOT` as needed for an installed RMM or `CPM_RMM_SOURCE` for a source RMM.

Reply:
I have some questions about this. You aren't wrong but also I need to get more specific. I'll post to the python chats here in a few and see who has an opinion and then resolve this accordingly.

Reply:
👍 would suggest posting here instead of Slack as the RMM CMake has gone through a lot of CMake work lately from a non-NVIDIAN 😄

Reply:
I'm gonna push a PR for `rmm` here in a few. Simple stuff, it fails with CPM now because of outdated links to the `cudf` `GTEST` and `benchmark` modules. Otherwise I think this problem is solved. Update coming in a few minutes.

Reply:
It should now be possible to just do `CPMFindPackage(...)` and then `target_link_libraries(tgt (PRIVATE|PUBLIC) rmm::rmm`. RMM is header-only, so this won't actually link anything, but it'll add the proper include paths for RMM.